bin/cylc-check-software

```diff
@@ -32,10 +32,12 @@ Arguments:
                 valid module arguments (lower-case equivalents accepted).
 """
 
-import sys
-import re
 import os
-from subprocess import check_call, PIPE, Popen, CalledProcessError
+import re
+import sys
+from subprocess import PIPE, CalledProcessError, check_call
+
+from cylc.sprocess import pcylc
```

Comment:
I don't like these names (sorry!) - they're not very descriptive, particularly in the case of "pcylc" - hard to guesss what that does without inspecting the function def.  What about `cylc_subprocess` and `cylc_popen`?

Reply:
Or `CylcPopen`.

Reply:
The names were changed as the static analysis uses a string comparison looking for popen and subprocess,  using those terms will require an additional #nosec tag at those lines instead of using the wrapper function tag.

Reply:
Ah, got it.  So we're having to compromise our code quality to satisfy a deficient static analysis tool. (Not your fault). Can we get away with something similar to the original but still understandable, e.g. `cylc_subproc` and `CylcPropen`?

Reply:
I'll test locally with bandit and refactor

Reply:
using cylc_subproc and procopen works

Reply:
thanks!